lib/fb/page.rb

```diff
@@ -30,12 +35,9 @@ def initialize(options = {})
     #   @name="page_fans", @description="Weekly: The
     #   number of new people who have liked your Page (Unique Users)",
     #   @value=123>,..}
-    def insights
+    def insights(options = {})
       @insights ||= begin
-        response_body = Fb::Request.new(
-          path: "/v2.9/#{@id}/insights",
-          params: insights_params).run
-        response_body["data"].map do |metric_data|
+        fetch_insights(options)["data"].map do |metric_data|
```

Comment:
should I validate options before calling fetch_insights(options)? There are no validation or exception handling in this class atm.